src/page/TaskPage/TaskPage.jsx

```diff
@@ -0,0 +1,72 @@
+import './TaskPage.scss'
+import { useState } from 'react';
+import { useParams } from "react-router"
+import { useNavigate } from 'react-router-dom';
+import { Header } from "../../components/Header/Header";
+import { useUserData } from "../../features/userData/useUserData.js";
+import { useTasks } from '../../features/tasks/useTasks';
+import { store } from '../../redux/store';
+
+export const TaskPage = () => {
+    const { id } = useParams();
+    const { tasks } = useTasks()
+
+    const taskToUpdate =  tasks && tasks.find(task => task.id === id)
+
+    const { userData } = useUserData()
+    const [update] = useState(id === 'create' ? false : true)
+    const [task, setTask] = useState(update ? taskToUpdate.task : '' )
+    const [date, setDate] = useState(update ? taskToUpdate.date : getDate())
```

Comment:
Тебе не нужен update в state для этого
```suggestion
    const isUpdate = id !== 'create'
    const [task, setTask] = useState(isUpdate ? taskToUpdate.task : '' )
    const [date, setDate] = useState(isUpdate ? taskToUpdate.date : getDate())
```